relayer/provider/cosmos/provider.go

```diff
@@ -1128,72 +1137,81 @@ func (cc *CosmosProvider) relayPacketsFromResultTx(ctx context.Context, src, dst
 			case !rp.pass:
 				rcvPackets = append(rcvPackets, rp)
 			}
-		}
-	}
 
-	// If there is a relayPacket, return it
-	if len(rcvPackets)+len(timeoutPackets) > 0 {
-		return rcvPackets, timeoutPackets, nil
+			// If there is a relayPacket, return it
+			if len(rcvPackets)+len(timeoutPackets) > 0 {
+				return rcvPackets, timeoutPackets, nil
+			}
+		}
 	}
 
 	return nil, nil, fmt.Errorf("no packet data found")
 }
 
 // acknowledgementsFromResultTx looks through the events in a *ctypes.ResultTx and returns
 // relayPackets with the appropriate data
-func acknowledgementsFromResultTx(dstChanId, dstPortId, srcChanId, srcPortId string, res *ctypes.ResultTx) ([]provider.RelayPacket, error) {
+func acknowledgementsFromResultTx(dstChanId, dstPortId, srcChanId, srcPortId string, resp *provider.RelayerTxResponse) ([]provider.RelayPacket, error) {
 	var ackPackets []provider.RelayPacket
-	for _, e := range res.TxResult.Events {
-		if e.Type == waTag {
+
+	rp := &relayMsgPacketAck{pass: false}
+	for key, attrVal := range resp.Events {
+		rp.pass = false
+		eventType, attrKey := splitEventKey(key)
+
+		if eventType == waTag {
 			// NOTE: Src and Dst are switched here
-			rp := &relayMsgPacketAck{pass: false}
-			for _, p := range e.Attributes {
-				if string(p.Key) == srcChanTag {
-					if string(p.Value) != srcChanId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == dstChanTag {
-					if string(p.Value) != dstChanId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == srcPortTag {
-					if string(p.Value) != srcPortId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == dstPortTag {
-					if string(p.Value) != dstPortId {
-						rp.pass = true
-						continue
-					}
-				}
-				if string(p.Key) == ackTag {
-					rp.ack = p.Value
+			if attrKey == srcChanTag {
```

Comment:
It looks like this sequence of if-statements can be one `switch attrKey` since we expect at most one branch to match.